packages/core/src/lib/store/index.ts

```diff
@@ -416,8 +416,18 @@ export async function createCursor(message: IDecodedMessage): Promise<Cursor> {
   }
 
   const contentTopicBytes = utf8ToBytes(message.contentTopic);
+  const pubSubTopicBytes = utf8ToBytes(message.pubsubTopic);
 
-  const digest = sha256(concat([contentTopicBytes, message.payload]));
+  let metaBytes: Uint8Array;
+  if (message.meta instanceof Uint8Array) {
```

Comment:
```suggestion
  if (message.meta && message.meta.length) {
```

Why checking with `instanceof`? What else could it be? I guess possibly a `Buffer` if application use NodeJS and in this instance we get a bug.

Reply:
> Why checking with `instanceof`? What else could it be? I guess possibly a `Buffer` if application use NodeJS and in this instance we get a bug.

nice catch, I am a JS illiterate!
How about also checking if the passed value is indeed a buffer? coz here `if (message.meta && message.meta.length)` `string` and `array` also have the length property?
May be something like this:

```  
     if (message.meta instanceof Uint8Array) {
        metaBytes = message.meta;
     } else if (Buffer.isBuffer(message.meta)) {
        metaBytes = new Uint8Array(message.meta);
     } else {
        metaBytes = new Uint8Array(); // default value
     }
```

Reply:
> you're introducing `Buffer`, an nodejs type, into an environment for browsers. ref: [libp2p/js-libp2p#2146 (comment)](https://github.com/libp2p/js-libp2p/issues/2146#event-10741737087)

So in short if an application uses NodeJS, a bug is fine from our end?

Reply:
My suggested change is enough.

The type of `message.meta` does not matter for this function, as long as it can be taken as input for `concat` because what is set/used (ie, the output of this function) is the output of `sha256`.
Hence, no need to try to wrapper any possible `Buffer` object.

There won't be a bug for NodeJS either.